File: app/src/main/java/com/sqube/tipshub/NewsStoryActivity.java
package com.sqube.tipshub;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ProgressBar;

public class NewsStoryActivity extends AppCompatActivity {

    private String url;
    private String SAVED_URL = "saved_url";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle savedInstanceS = savedInstanceState;
        if(savedInstanceState!=null)
            url = savedInstanceState.getString(SAVED_URL);
        else
            url = getIntent().getStringExtra("url");
        setContentView(R.layout.activity_news_story);
        WebView webView = findViewById(R.id.wvNews);
        ProgressBar progressBar = findViewById(R.id.progressBar);

        Log.i("News", "onCreate: "+url);
        webView.getSettings().setBuiltInZoomControls(true);
        webView.getSettings().setDisplayZoomControls(false);
        webView.getSettings().setUseWideViewPort(true);
        webView.getSettings().setDomStorageEnabled(true);
        webView.getSettings().setPluginState(WebSettings.PluginState.ON);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.getSettings().setSupportMultipleWindows(true);

        webView.setWebViewClient(new WebViewClient(){

            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                progressBar.setVisibility(View.GONE);
            }
        });
        webView.loadUrl(url);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        outState.putString(SAVED_URL, url);
        super.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/com/sqube/tipshub/NewsStoryActivity.kt
package com.sqube.tipshub

import android.os.Bundle
import android.util.Log
import android.view.View
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import com.sqube.tipshub.databinding.ActivityNewsStoryBinding

class NewsStoryActivity : AppCompatActivity() {
    private var url: String? = null
    private val savedUrl = "saved_url"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        url = if (savedInstanceState != null) savedInstanceState.getString(savedUrl) else intent.getStringExtra("url")
        val binding = ActivityNewsStoryBinding.inflate(layoutInflater)
        setContentView(binding.root)
        Log.i("News", "onCreate: $url")
        with(binding){
            wvNews.settings.builtInZoomControls = true
            wvNews.settings.displayZoomControls = false
            wvNews.settings.useWideViewPort = true
            wvNews.settings.domStorageEnabled = true
            wvNews.settings.pluginState = WebSettings.PluginState.ON
            wvNews.settings.javaScriptEnabled = true
            wvNews.settings.setSupportMultipleWindows(true)
            wvNews.webViewClient = object : WebViewClient() {
                override fun onPageFinished(view: WebView, url: String) {
                    super.onPageFinished(view, url)
                    progressBar.visibility = View.GONE
                }
            }
            wvNews.loadUrl(url!!)
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putString(savedUrl, url)
        super.onSaveInstanceState(outState)
    }
}